ggml/src/ggml-cpu/ggml-cpu-aarch64.cpp

```diff
@@ -3641,14 +3677,14 @@ static block_q4_0x8 make_block_q4_0x8(block_q4_0 * in, unsigned int blck_size_in
     return out;
 }
 
-static int repack_q4_0_to_q4_0_4_bl(struct ggml_tensor * t, int interleave_block, const void * restrict data, size_t data_size) {
+static int repack_q4_0_to_q4_0_4_bl(struct ggml_tensor * t, int interleave_block, const void * GGML_RESTRICT data, size_t data_size) {
     GGML_ASSERT(t->type == GGML_TYPE_Q4_0);
     GGML_ASSERT(interleave_block == 4 || interleave_block == 8);
 
     block_q4_0x4 * dst = (block_q4_0x4 *)t->data;
     const block_q4_0 * src = (const block_q4_0 *)data;
     block_q4_0 dst_tmp[4];
-    int nrow = t->ne[1]; // Number of rows
+    int nrow = t->ne[1]*t->ne[2]*t->ne[3]; // Number of rows
```

Comment:
```suggestion
    int nrow = ggml_nrows(t);
```

Btw, I'm not very confident that the repack/gemm logic would work correctly for `(t->ne[1]%4 != 0) && (t->ne[2]*t->ne[3] > 1)`. Maybe it's better to assert `t->ne[2]*t->ne[3] == 1` for now and re-enable when we can test this.

Reply:
It is needed for  GGML_OP_MUL_MAT_ID  (where t->ne[2] > 1) and look to work.

but it is not use with GGML_OP_MUL_MAT

Reply:
`(t->ne[1]%4 != 0)`

I'll have a look at this one. It did not remove existing control, but I don't remember if it's controlled elsewhere.